lib/metrics/prometheus.js

```diff
@@ -34,23 +37,33 @@ class PrometheusMetric {
         }
     }
 
+    _normalize(str) {
+        return String(str).replace( /[^a-z0-9]/g, '_' ) // replace non-alphanumerics
+            .replace( /_+/g, '_' ) // dedupe underscores
+            .replace( /(^_+|_+$)/g, '' ); // trim leading and trailing underscores
+    }
+
     increment(amount, labels) {
         this._handleStaticLabels(labels);
+        labels.map(this._normalize);
         this.metric.labels.apply(this.metric, labels).inc(amount);
     }
 
     decrement(amount, labels) {
         this._handleStaticLabels(labels);
+        labels.map(this._normalize);
```

Comment:
You're not assigning the result to labels. .map does not modify the array itself afaik.